woocommerce/woocommerce-ios
# Coupons: Usage Details screen (read only)

Closes: #5766 


### Description

This PR adds the usage details screen that can be navigated from the coupon details screen. The fields on the screen are currently not editable since we support viewing coupon details only in Milestone 1. Changes include:
- Update the Coupon Details view to navigate to the Usage Details upon tapping on the Usage Details row.
- Add UI and populate details on the Usage Details screen.
- Update some reusable views:
  - TitleAndValueRow to have a custom initializer and default values for `bold`, `selectable` and `action`.
  - TitleAndTextFieldRow to have a new parameter `editable` in its initializer.

### Design Reviews
To design team: I tweaked this screen a bit to match with the goal of Milestone 1, which is to view coupon details only:
- The text fields and toggles cannot be edited
- A missing field for Usage Limit Per User is added
- Subtitles in the Usage Limits section are not present for now since the toggles are disabled.
- Also: I'm leaving the Exclusions section out of this screen following the comment: p1644401679450069-slack-C02TD63FZA8. Let me know if we should put it back.

### Testing instructions

- Make sure that you have at least one coupon on your test store. Otherwise, navigate to WP Admin > Marketing > Coupons to create one.
- On the app, make sure that coupon management is enabled in Menu > Settings > Experimental Features.
- On the Menu tab, select Coupons and then select a coupon in the list and choose Usage Details. Notice that the information is displayed correctly and not editable.

### Screenshots

<img src="https://user-images.githubusercontent.com/5533851/154198202-3b0f5f46-dc9b-4edc-a224-e7cfccf9206a.png" width=320 />

---
